libraries/botbuilder-dialogs-adaptive/src/adaptiveDialog.ts

```diff
@@ -1038,174 +1014,314 @@ export class AdaptiveDialog<O extends object = {}> extends DialogContainer<O> im
+            const expectedOnly = propSchema.expectedOnly ?? globalExpectedOnly;
+
+            for (const propEntity of propSchema.entities) {
+                const entityName = this.stripProperty(propEntity);
+                if (entities[entityName] && (isExpected || !expectedOnly.includes(entityName))) {
+                    for (const entity of entities[entityName]) {
+                        if (!entity.property) {
+                            assignments.push(
+                                new EntityAssignment({
+                                    value: entity,
+                                    property: propSchema.name,
+                                    operation: entity.operation,
+                                    isExpected,
+                                })
+                            );
+                        } else if (entity.property === entityName && !entity.value && !entity.operation && isExpected) {
+                            // Recast property with no value as match for property entities.
+                            assignments.push(
+                                new EntityAssignment({
+                                    value: entity,
+                                    property: propSchema.name,
+                                    operation: null,
+                                    isExpected,
+                                })
+                            );
                         }
-                    } else {
-                        infos.push(info);
                     }
                 }
             }
-        }
-    }
+        });
 
-    /**
-     * @private
-     */
-    private candidates(entities: NormalizedEntityInfos, expected: string[]): Partial<EntityAssignment>[] {
-        const candidates: Partial<EntityAssignment>[] = [];
-        const globalExpectedOnly: string[] = this.dialogSchema.schema[this.expectedOnlyKey] || [];
-        const usedEntityType = new Set<string>([this.utteranceKey]);
-        const usedEntity: Map<string, Partial<EntityInfo>> = new Map();
-
-        // Emit entities that already have a property
-        for (const entityName in entities) {
-            const alternatives = entities[entityName];
-            for (const alternative of alternatives) {
-                if (alternative.property) {
-                    usedEntity.set(alternative.name, alternative);
-                    candidates.push({
-                        entity: alternative,
-                        property: alternative.property,
-                        operation: alternative.operation,
-                        isExpected: expected.indexOf(alternative.property) >= 0,
-                    });
+        // Add default operations.
+        for (const assignment of assignments) {
+            if (!assignment.operation) {
+                // Assign missing operation.
+                if (lastEvent == AdaptiveEvents.chooseEntity && assignment.value.property === nextAssignment.property) {
```

Comment:
`===` (you could search + replace in this file if there are any others that don't involve `null` or `undefined`)